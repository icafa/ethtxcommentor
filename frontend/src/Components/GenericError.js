import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { EmptyStateCard, IconError } from '@aragon/ui'

const Wrapper = styled.div`
  display: flex;
  height: 100vh;
  width: 100vw;
  align-items: center;
  justify-content: center;
`

const GenericError = ({ reportCallback }) => (
  <Wrapper>
    <EmptyStateCard
      title="An unexpected error has ocurred"
      text="We're sorry for the inconvenience."
      icon={<IconError />}
      actionText="Report Error"
      onActivate={reportCallback}
    />
  </Wrapper>
)

GenericError.propTypes = {
  reportCallback: PropTypes.func.isRequired,
}

export default GenericError
