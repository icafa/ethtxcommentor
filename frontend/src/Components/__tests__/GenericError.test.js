import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import GenericError from '../GenericError'
import { Main } from '@aragon/ui'

describe('GenericError Component Tests', () => {
  it('matches the snapshot', () => {
    const reportCallback = jest.fn()
    const { asFragment } = render(
      <Main>
        <GenericError reportCallback={reportCallback} />
      </Main>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders in the dom succesfully and shows static data', () => {
    const reportCallback = jest.fn()
    const { getByText } = render(
      <Main>
        <GenericError reportCallback={reportCallback} />
      </Main>
    )
    const title = getByText(/an unexpected error has ocurred/i)
    expect(title).toBeInTheDocument()
    const info = getByText(/we're sorry for the inconvenience/i)
    expect(info).toBeInTheDocument()
  })
  it('succesfully calls the reportCallback prop passed', () => {
    const reportCallback = jest.fn()
    const { getByText } = render(
      <Main>
        <GenericError reportCallback={reportCallback} />
      </Main>
    )
    const reportButton = getByText(/report error/i)
    fireEvent.click(reportButton)
    expect(reportCallback).toHaveBeenCalledTimes(1)
  })
})
