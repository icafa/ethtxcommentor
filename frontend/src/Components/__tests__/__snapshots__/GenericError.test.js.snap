// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericError Component Tests matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-cHGsZl enfjgR"
    >
      <div
        class="Card__StyledCard-sc-13r75gj-0 EmptyStateCard__Main-ov2yly-0 dwLKdW"
      >
        <section>
          <svg
            height="22"
            viewBox="0 0 22 22"
            width="22"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              transform="translate(4 4)"
            >
              <rect
                fill="#FF445D"
                height="14"
                rx="7"
                width="14"
              />
              <path
                d="M6.155 8.547h1.298V3.3H6.155v5.247zM6.045 11h1.529V9.537H6.045V11z"
                fill="#FFF"
              />
            </g>
          </svg>
          <h1
            class="EmptyStateCard__Heading-ov2yly-2 dmyaZr"
          >
            <span
              class="Text-yxw4o9-0 hUTBun"
              color="#00CBE6"
            >
              An unexpected error has ocurred
            </span>
          </h1>
          <div
            class="Text-yxw4o9-0 hrwmLl"
          >
            We're sorry for the inconvenience.
          </div>
          <div
            class="EmptyStateCard__ActionWrapper-ov2yly-3 cWXFuk"
          >
            <button
              class="Button__StyledButton-sc-8npd5h-0 VDYDq EmptyStateCard__DefaultActionButton-ov2yly-1 PmJbu"
              mode="strong"
              type="button"
            >
              Report Error
            </button>
          </div>
        </section>
      </div>
    </div>
    <div
      class="ToastHub__Container-sc-1y0i8xl-0 stXpK"
    />
  </div>
</DocumentFragment>
`;
